File: Crio/crio_programming_interview_problems-master/AddTwoNumbers/AddTwoNumbers.py
from Solution import *
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def createList(numbers):
	if(len(numbers) == 0):
		return None

	head = ListNode(numbers[0])
	node = head
	for i in range(1, len(numbers)):
		node.next = ListNode(numbers[i])
		node = node.next

	return head

def extractList(head):
	arr = []

	while(head):
		arr.append(head.val)
		head = head.next

	return arr


if __name__ == '__main__':
    n = int(input())
    nums1 = input().split()
    nums1 = [int(i) for i in nums1]
    m = int(input())
    nums2 = input().split()
    nums2 = [int(i) for i in nums2]
    l1 = createList(nums1)
    l2 = createList(nums2)
    resultList = addTwoNumbers(l1, l2)
    result = extractList(resultList)
    for i in result:
        print(i,end=' ')

File: Crio/crio_programming_interview_problems-master/AddTwoNumbers/AddTwoNumbers.java
import java.io.*;
import java.util.*;
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

class AddTwoNumbers {
    public static ListNode createList(List<Integer> numbers) {
        if (numbers.size() == 0)
            return null;
    
        ListNode head = new ListNode(numbers.get(0));
        ListNode node = head;
    
        for (int i = 1; i < numbers.size(); ++i) {
            node.next = new ListNode(numbers.get(i));
            node = node.next;
        }
    
        return head;
    }
    
    public static List<Integer> extractList(ListNode head) {
        List<Integer> lst = new ArrayList<Integer>();
    
        while(head != null) {
            lst.add(head.val);
            head = head.next;
        }
    
        return lst;
    }
    
    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        List<Integer> firstNumber = new ArrayList<Integer>();
        int n = scanner.nextInt();
        for (int i = 0; i < n; ++i)
            firstNumber.add(scanner.nextInt());

        List<Integer> secondNumber = new ArrayList<Integer>();
        n = scanner.nextInt();
        for (int i = 0; i < n; ++i)
            secondNumber.add(scanner.nextInt());

        scanner.close();

        ListNode firstNumberList = createList(firstNumber);
        ListNode secondNumberList = createList(secondNumber);

        ListNode addedNumber = new Solution().addTwoNumbers(firstNumberList, secondNumberList);
        List<Integer> result = extractList(addedNumber);
        for(Integer i : result){
            System.out.print(i+" ");
        }
    }
}